$OpenBSD: patch-src_gmtk_media_player_h,v 1.6 2013/07/12 09:58:56 dcoppa Exp $

Use case-insensitive parsing of metadata attributes
(upstream svn revision r212)

If the user gives some mplayer video output driver (vo) options in
the "Video Output" preferences they should be passed along on the
mplayer command line (upstream svn revision r222)

--- src/gmtk_media_player.h.orig	Tue Feb 19 18:02:13 2013
+++ src/gmtk_media_player.h	Thu Jul 11 17:03:07 2013
@@ -182,7 +182,8 @@ typedef enum {
     ATTRIBUTE_PROFILE,
     ATTRIBUTE_PREFERRED_AUDIO_LANGUAGE,
     ATTRIBUTE_PREFERRED_SUBTITLE_LANGUAGE,
-    ATTRIBUTE_SPEED_SET
+    ATTRIBUTE_SPEED_SET,
+    ATTRIBUTE_GENRE
 } GmtkMediaPlayerMediaAttributes;
 
 typedef enum {
@@ -327,6 +328,7 @@ struct _GmtkMediaPlayer {
     gchar *artist;
     gchar *title;
     gchar *album;
+    gchar *genre;
     gboolean enable_divx;
     gboolean disable_xvmc;
     gboolean retry_on_full_cache;
@@ -398,6 +400,13 @@ struct _GmtkMediaPlayer {
 
     gboolean disposed;
     GMutex *player_lock;
+    GRegex *name_regex;
+    GRegex *genre_regex;
+    GRegex *title_regex;
+    GRegex *artist_regex;
+    GRegex *album_regex;
+    GRegex *deintN_regex;
+
 };
 
 struct _GmtkMediaPlayerClass {
